public/app/features/dashboard/components/GenAI/GenAIHistory.tsx

```diff
@@ -0,0 +1,141 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useState } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Button, Divider, HorizontalGroup, Icon, Input, Link, Spinner, Text, useStyles2 } from '@grafana/ui';
+
+import { getFeedbackMessage } from './GenAIPanelTitleButton';
+import { GenerationHistoryCarousel } from './GenerationHistoryCarousel';
+import { QuickActions } from './QuickActions';
+import { useOpenAIStream } from './hooks';
+import { Message, OPEN_AI_MODEL, QuickFeedback } from './utils';
+
+export interface GenAIHistoryProps {
+  history: string[];
+  messages: Message[];
+  onApplySuggestion: (suggestion: string) => void;
+  updateHistory: (historyEntry: string) => void;
+}
+
+const temperature = 0.5;
+
+export const GenAIHistory = ({ history, messages, onApplySuggestion, updateHistory }: GenAIHistoryProps) => {
+  const styles = useStyles2(getStyles);
+
+  const [currentIndex, setCurrentIndex] = useState(1);
+  const [response, setResponse] = useState<string>('');
+
+  const { setMessages, reply, isGenerating } = useOpenAIStream(OPEN_AI_MODEL, temperature);
+
+  // @TODO Duplicate, find a better way to do this
+  useEffect(() => {
+    if (reply !== '') {
+      setResponse(reply.replace(/^"|"$/g, ''));
+    }
+  }, [reply]);
+
+  useEffect(() => {
+    if (response !== '' && !isGenerating) {
+      setResponse('');
+      setCurrentIndex(1);
+    }
+  }, [response, isGenerating]);
+
+  const onSubmit = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      onGenerateWithFeedback(event.currentTarget.value as QuickFeedback);
```

Comment:
This seems to be a completely wrong type assertion? Or is there something I'm missing?

It looks like `event.currentTarget.value` is any arbitary string the user typed in, not only a string contained in the `QuickFeedback` enum.

Reply:
yes - good catch! forgot to update the method.